src/App.js

```diff
@@ -9,23 +9,41 @@ export default class App extends Component {
   constructor() {
     super();
     this.state = {
-      movies: movieData.movies
+      movies: [],
+      error: ''
     }
   }
-
-  displayMovieDetails = (id) => {
-    const foundMovie = this.state.movies.filter(movie => movie.id === id);
-    this.setState({movies: foundMovie})
+  
+  componentDidMount = async() => {
+    let fetchedMovies = [];
+    try {
+        fetchedMovies = await getAllMovies();
+        this.setState({movies: fetchedMovies.movies})
+      } catch (e) {
+        this.setState({error: e.message})
+      }
   }
 
+  displayMovieDetails = async id => {
+    let fetchedMovie = [];
```

Comment:
Same comment as before - you can define `fetchedMovie` inside the try block, because nothing outside the try block needs to access the variable.